client/live_test.go

```diff
@@ -45,7 +46,8 @@ func (s *LiveTests) TestAuthenticateFail(c *gc.C) {
 	osclient := client.NewClient(&cred, s.authMode, nil)
 	c.Assert(osclient.IsAuthenticated(), gc.Equals, false)
 	err := osclient.Authenticate()
-	c.Assert(err, gc.ErrorMatches, "authentication failed(\n|.)*")
+	isAuthorised := gooseerrors.IsUnauthorised(err)
+	c.Assert(isAuthorised, gc.Equals, true)
```

Comment:
We probably need a test where a non auth error like failure to make service urls or whatever is checked so that isAuthorised = false